osu.Game/Online/Leaderboards/LeaderboardScoreV2.cs

```diff
@@ -0,0 +1,686 @@
+                        CornerRadius = corner_radius,
+                        Children = new Drawable[]
+                        {
+                            totalScoreBackground = new Box
+                            {
+                                RelativeSizeAxes = Axes.Both,
+                                Colour = totalScoreBackgroundGradient,
+                            },
+                            new Box
+                            {
+                                RelativeSizeAxes = Axes.Both,
+                                Colour = ColourInfo.GradientHorizontal(backgroundColour.Opacity(0), OsuColour.ForRank(score.Rank).Opacity(0.5f)),
+                            },
+                            new FillFlowContainer
+                            {
+                                AutoSizeAxes = Axes.Both,
+                                Anchor = Anchor.CentreLeft,
+                                Origin = Anchor.CentreLeft,
+                                Direction = FillDirection.Vertical,
+                                Padding = new MarginPadding { Horizontal = corner_radius },
+                                Children = new Drawable[]
+                                {
+                                    scoreText = new OsuSpriteText
+                                    {
+                                        Anchor = Anchor.TopRight,
+                                        Origin = Anchor.TopRight,
+                                        UseFullGlyphHeight = false,
+                                        Shear = -shear,
+                                        Current = scoreManager.GetBindableTotalScoreString(score),
+                                        Font = OsuFont.GetFont(size: 30, weight: FontWeight.Light),
+                                    },
+                                    modsContainer = new FillFlowContainer<ColouredModSwitchTiny>
+                                    {
+                                        Anchor = Anchor.TopRight,
+                                        Origin = Anchor.TopRight,
+                                        Shear = -shear,
+                                        AutoSizeAxes = Axes.Both,
+                                        Direction = FillDirection.Horizontal,
+                                        ChildrenEnumerable = score.Mods.AsOrdered().Select(mod => new ColouredModSwitchTiny(mod) { Scale = new Vector2(0.375f) })
+                                    },
+                                    modsCounter = new OsuSpriteText
+                                    {
+                                        Anchor = Anchor.TopRight,
+                                        Origin = Anchor.TopRight,
+                                        Shear = -shear,
+                                        Text = $"{score.Mods.Length} mods",
+                                        Alpha = 0,
+                                    }
```

Comment:
I have [stirred up](https://discord.com/channels/188630481301012481/188630652340404224/1239900324915974255) some discussion on whether this can be replaced by an indicator next to an arbitrary subset of mods, and it was concluded that it's better. For example (design is arbitrary):

![image](https://github.com/ppy/osu/assets/22781491/dba7b422-1421-4b21-84f9-02336e861d31)

Currently [waiting on response](https://discord.com/channels/188630481301012481/188630652340404224/1239908144482881546) from @arflyte for a better design of this indicator, unless it's fine to proceed with the current one (I definitely need to rewrite the code because I faked up a `Mod` to achieve this).

Reply:
Added in https://github.com/ppy/osu/pull/22237/commits/59553780040dbf82c48b7bd675f0c64ba97194bd